zend_abstract_interface/interceptor/php7/interceptor.c

```diff
@@ -176,7 +176,15 @@ static inline int zai_interceptor_ext_nop_handler_no_prev(zend_execute_data *exe
         }
     }
 
-    return ZEND_USER_OPCODE_DISPATCH;
+    if (&execute_data->func->op_array == op_array) {
+        return ZEND_USER_OPCODE_DISPATCH;
+    } else {
+        // the code was changed, so instead of executing the original handler of
+        // opline->opcode (gotten via zend_vm_get_opcode_handler_func),
+        // we return ZEND_USER_OPCODE_CONTINUE so that user opcode handler
+        // of the new execute_data->opline is executed
+        return ZEND_USER_OPCODE_CONTINUE;
```

Comment:
Let's move that check to after zai_hook_memory_table_insert to avoid changing the hot path.